src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,49 @@
 package core.basesyntax;
 
-class Pair {
+class Pair<T,A> {
+    private T one;
+    private A two;
 
+    private Pair(T one, A two) {
+        this.one = one;
+        this.two = two;
+    }
+
+    public T getFirst() {
+        return one;
+    }
+
+    public A getSecond() {
+        return two;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+
+        if (o == null) {
+            return false;
+        }
+        if (o.getClass().equals(getClass())) {
+            Pair<?, ?> pair = (Pair<?, ?>) o;
+            if (pair.one == null && one == null && pair.two == null && two == null) {
+                return one == pair.one && two == pair.two;
+            }
+            return one.equals(pair.one) && two.equals(pair.two);
```

Comment:
what happens on 34 row if pair.one != null  and one == null? 
try to optimize your equals() method